bitcoin/src/blockdata/transaction.rs

```diff
@@ -1141,6 +1143,38 @@ mod sealed {
     impl Sealed for super::Version {}
 }
 
+#[cfg(feature = "arbitrary")]
+impl<'a> Arbitrary<'a> for InputWeightPrediction {
+    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+        // limit script size to 4Mwu block size.
+        let max_block = Weight::MAX_BLOCK.to_wu() as usize;
+        let input_script_len = u.int_in_range(0..=max_block)?;
+        let remaining = max_block - input_script_len;
+
+        // create witness data if there is remaining space.
+        let mut witness_length = u.int_in_range(0..=remaining)?;
+        let mut witness_element_lengths = Vec::new();
+
+        // build vec of random witness element lengths.
+        while witness_length > 0 {
+            let elem = u.int_in_range(1..=witness_length)?;
```

Comment:
IIRC 0-sized elements are valid. Also IIRC the limit for one element is 520B, so it might be used here.

Reply:
Fair point.  The reason this is from `1` and not `0` is because it will quickly become a stack overflow since selecting `0` infinitely cases an infinite loop.  Although, limiting the number of possible witness elements would be a better way to handle this if `0` is possible.

Also, this will create a vector where the largest are always first which isn't truly a random distribution.

Reply:
You can still subtract 1 from the max len. Actually, if you want to use max len then you need to account for element len serialization which is 1 to 9 bytes (1 to 3 in practice).